### PR TITLE
DSPDC-903 Add Helm chart for Argo template

### PR DESCRIPTION
**Pull request by danxmoran**
It makes sense to me to co-locate the Helm chart for this tool in the same repo as the tool, so they can be versioned in lock-step.

### CODE REVIEW

File: argo/templates/template.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{{- $version := default "latest" .Chart.AppVersion }}
+apiVersion: argoproj.io/v1alpha1
```
Comment (danxmoran) [inline argo/templates/template.yaml:2]: For reference, here's the version in ClinVar: https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/templates/extract-xml-to-json.yaml

I'll comment where I changed things

File: argo/templates/template.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    - name: main
+      inputs:
+        parameters:
+          - name: input-pvc-name
+          - name: input-xml-path
+          - name: output-pvc-name
+          - name: objects-per-part
+          - name: gunzip
+          - name: memory-mib
+          - name: cpu-m
```
Comment (danxmoran) [inline argo/templates/template.yaml:17]: I added more parameters here for all the things we might want to toggle

File: argo/templates/template.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        image: us.gcr.io/broad-dsp-gcr-public/xml-to-json-list-clp:{{ $version }}
+        volumeMounts:
+          - name: in
+            mountPath: /in
+            readOnly: true
+          - name: out
+            mountPath: /out
+        {{- $mem := "{{inputs.parameters.memory-mib}}" }}
+        {{- $cpu := "{{inputs.parameters.cpu-m}}" }}
+        podSpecPatch: |
```
Comment (danxmoran) [inline argo/templates/template.yaml:36]: This had to become a `podSpecPatch` to support parameters in the resource specs

File: argo/templates/template.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          set -euo pipefail
+
+          declare -a args=(
+            --input=${INPUT}
+            --output=/out
+            --objects-per-part=${OBJS_PER_PART}
+          )
+          if [[ "${GUNZIP}" == "true" ]]; then
+            args+=(--gunzip-input)
+          fi
```
Comment (danxmoran) [inline argo/templates/template.yaml:66]: This had to become a `script` to support the optional flag here.

File: build.sbt [hunk 5]
```diff
@@ -13,6 +13,10 @@ val scalaTestVersion = "3.1.1"
 
 lazy val `monster-xml-to-json-list` = project
   .in(file("."))
+  .aggregate(`xml-to-json-list-clp`, `xml-to-json-list-template`)
+
+lazy val `xml-to-json-list-clp` = project
+  .in(file("clp"))
```
Comment (danxmoran) [inline build.sbt:19]: Making the build multi-project was the simplest way to enable using the new plugin. The downside is I had to move all the existing code into a sub-directory.

File: build.sbt [hunk 6]
```diff
@@ -35,3 +39,11 @@ lazy val `monster-xml-to-json-list` = project
       "org.scalatest" %% "scalatest" % scalaTestVersion
     ).map(_ % Test)
   )
+
+  lazy val `xml-to-json-list-template` = project
+    .in(file("argo"))
```
Comment (danxmoran) [inline build.sbt:44]: Is "argo" sufficient? Should I mention "chart" or "helm" in the name anywhere?

Comment (raaidbroad) [reply build.sbt:44]: I think argo is sufficient

File: argo/values.schema.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "templateName": {
+      "description": "Name to assign to the generated WorkflowTemplate object",
+      "type": "string"
+    }
+  },
+  "required": ["name"]
```
Comment (raaidbroad) [inline argo/values.schema.json]: should this be templateName?

**Review by raaidbroad — COMMENTED** [review]
Looks good, made a comment or two

File: argo/values.schema.json (hunk 7 above)
Comment (danxmoran) [reply argo/values.schema.json]: Yes, good catch!

Comment (danxmoran) [reply argo/values.schema.json]: If only we had a schema for our schema...

**Review by raaidbroad — APPROVED** [review]

**Review by snwessel — APPROVED** [review]

**Pull request merged**